modules/openapi-generator/src/main/resources/python/python-experimental/model_utils.mustache

```diff
@@ -622,6 +691,56 @@ def deserialize_primitive(data, klass, path_to_item):
         )
 
 
+def get_discriminator_class(model_class,
+                            discr_name,
+                            discr_value, cls_visited):
+    """Returns the child class specified by the discriminator.
+
+    Args:
+        model_class (OpenApiModel): the model class.
+        discr_name (string): the name of the discriminator property.
+        discr_value (any): the discriminator value.
+        cls_visited (list): list of model classes that have been visited.
+            Used to determine the discriminator class without
+            visiting circular references indefinitely.
+
+    Returns:
+        used_model_class (class/None): the chosen child class that will be used
+            to deserialize the data, for example dog.Dog.
+            If a class is not found, None is returned.
+    """
+
+    if model_class in cls_visited:
+        # The class has already been visited and no suitable class was found.
+        return None
+    cls_visited.append(model_class)
+    used_model_class = None
+    if discr_name in model_class.discriminator:
+        class_name_to_discr_class = model_class.discriminator[discr_name]
+        used_model_class = class_name_to_discr_class.get(discr_value)
```

Comment:
We are not yet handling the use case where discr_name is not in model_class.discriminator here.
If we hit that case, then the discriminator has changed and we did not find what we are looking for.
Here, can you add:
```
else:
    # the discr_name has changed and we did not find the class that we are looking for
    return None
```

Or maybe we should make it the clearer:
```
if discr_name not in model_class.discriminator:
    # the discr_name has changed and we did not find the class that we are looking for
    return None
class_name_to_discr_class = model_class.discriminator[discr_name]
used_model_class = class_name_to_discr_class.get(discr_value)
```